lib/dashing/app.rb

```diff
@@ -99,6 +99,22 @@ def protected!
   end
 end
 
+post '/widgets' do
+  request.body.rewind
+  body = JSON.parse(request.body.read)
+  auth_token = body.delete("auth_token")
+  if !settings.auth_token || settings.auth_token == auth_token
+    body["widgets"].each do |widget|
+      widget_id = widget.delete("id")
+      send_event(widget_id, widget)
+    end
+    204 # response without entity body
```

Comment:
This is the existing pattern in the file -- they should all be modified in this way if desired.

My preference is to leave it as a comment, so that you don't have to look up 204. IMHO it is not a very common HTTP response code.